packages/react-native-dogfood/src/components/VideoWrapper.tsx

```diff
@@ -22,19 +26,19 @@ export const VideoWrapper = ({ children }: PropsWithChildren<{}>) => {
     [username, userImageUrl],
   );
 
-  const tokenOrProvider = useCallback(async () => {
+  const tokenProvider = useCallback(async () => {
     const token = await createToken({ user_id: username });
     return token;
   }, [username]);
 
-  const videoClient = useCreateStreamVideoClient({
-    user,
-    tokenOrProvider: tokenOrProvider,
-    apiKey: STREAM_API_KEY,
-    options: {
-      preferredVideoCodec: Platform.OS === 'android' ? 'VP8' : undefined,
-    },
-  });
+  const [videoClient] = useState<StreamVideoClient>(
+    () =>
+      new StreamVideoClient({
+        apiKey: STREAM_API_KEY,
+        user,
+        tokenProvider,
+      }),
+  );
```

Comment:
Why does videoClient needs to be state? if instantiated once?

Reply:
Converted it to useRef

Reply:
Apologies for jumping in here but I think you should keep the client within a useState variable.

The reasoning: the client should be typically instantiated only once via `useState(() => client)` factory. In case of dependency change, there should be an `useEffect` that would dispose of the previous client instance, set up a new one and eventually update the state. In case you don't want to react on dependency changes, you can omit the `useEffect` part.

The solution with `useRef` causes new StreamVideoClient instances to be created on every render and then immediately discarded with no way to have them properly disposed of.